### DISCUSSION TITLE
图数据库优化问题

### DISCUSSION THREAD

**Discussion by waderwu** [BytecodeDL/ByteCodeDL#2] [General]
在使用中发现，如果数据量一大，查询的调用步数一多，就查不出结果了，不知道对图数据库优化能不能解决这个问题
优化的方向有这么几个
1. 把neo4j “优化” 掉，换成其他图数据库，主要考虑的是多跳的查询性能
2. 从建索引，优化查询语句，以空间换时间，修改neo4j配置等方面进行优化

有没有这方面有经验，或者对这感兴趣的同学，可以帮忙动手测试一下 🙌

**Comment by waderwu** [comment]
有一些图数据库厂商自己会发布一些测试报告，比如 https://www.galaxybase.com/report/developer 看着效果好像还可以

**Comment by waderwu** [comment]
《我在学习和实践图数据库 Neo4j 的漫漫成长路》https://vancir.github.io/the-journey-of-learning-neo4j.html 
这篇文章里面提到的优化建议应该有用

**Comment by waderwu** [comment]
昨天尝试了一天发现效果不明显，决定还是先引入最短路径，减少节点数量和环的出现，来临时解决问题。

**Comment by waderwu** [comment]
利用最短路径优化了一下，具体代码见分支 https://github.com/BytecodeDL/ByteCodeDL/tree/cha-optimization

**Comment by waderwu** [comment]
查找两点之间的所有路径是个NP hard问题，我觉得可以尝试使用启发式算法，返回有限条路径，有空的可以研究研究

**Comment by Y4er** [comment]
neo4j可以参考[tabby](https://github.com/wh1t3p1g/tabby/wiki/Neo4j%E7%8E%AF%E5%A2%83%E9%85%8D%E7%BD%AE#6-%E5%9B%BE%E6%95%B0%E6%8D%AE%E5%BA%93%E4%BC%98%E5%8C%96)来为neo4j建立索引的方案。
